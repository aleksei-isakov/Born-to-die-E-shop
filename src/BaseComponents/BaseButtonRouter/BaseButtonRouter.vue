<template>
  <button class="BaseButtonRouter" @click="click">
    <router-link :to="require(path)"><slot></slot></router-link>
  </button>
</template>

<script>
export default {
  name: 'BaseButtonRouter',

  props: {
    path: {
      type: String,
      default: ''
    },

    data() {
      return {};
    },

    methods: {
      click() {
        this.$emit('click', data);
      }
    }
  }
};
</script>
