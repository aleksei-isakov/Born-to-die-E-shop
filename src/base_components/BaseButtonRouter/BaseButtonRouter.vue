<template>
  <router-link :to="path">
    <slot></slot>
  </router-link>
</template>

<script>
export default {
  name: 'BaseButtonRouter',

  props: {
    path: {
      type: String,
      default: '',
      required: true
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/variables.scss';
.base-button-router {
  display: block;
  text-transform: uppercase;
  color: #e7e7e7;
  font-weight: 500;
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.03em;
  font-family: Roboto, sans-serif;
  transition: $transition;
  &:hover {
    text-decoration: none;
    color: $white;
  }
  &.router-link-exact-active {
    color: $white;
  }
}
</style>
