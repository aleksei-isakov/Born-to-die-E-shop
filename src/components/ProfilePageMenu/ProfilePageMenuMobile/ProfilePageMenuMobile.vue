<template>
  <div class="profile-page-menu__bottombar">
    <div v-for="item in items" :key="item.id" class="bottombar-menu-item">
      <base-button-router :path="item.path">
        <i :class="`${item.icon} bottombar-menu-item__icon fa-2x`" />
      </base-button-router>
      <base-button-router :path="item.path"
        >{{ item.title }}
      </base-button-router>
    </div>
  </div>
</template>

<script>
import { BaseButtonRouter } from '@/base_components/';

export default {
  name: 'ProfilePageMenuMobile',

  components: {
    BaseButtonRouter
  },

  data() {
    return {
      items: [
        {
          title: 'Profile',
          icon: 'fas fa-house-user',
          path: '/profile'
        },

        {
          title: 'Wishlist',
          icon: 'fas fa-heart',
          path: '/wishlist'
        },

        {
          title: 'Orders',
          icon: 'fas fa-folder',
          path: '/orders'
        },

        {
          title: 'Address book',
          icon: 'fas fa-address-book',
          path: '/address-book'
        }
      ]
    };
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.profile-page-menu {
  &__bottombar {
    position: absolute;
    bottom: 0;
    left: 0;
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    width: 100%;
    background: $primary;

    > .bottombar-menu-item {
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
      width: 25%;
      opacity: 50%;
      min-height: 80px;
      color: $white;
      &:hover {
        color: $white;
        opacity: 100%;
      }

      .base-button-router {
        color: $white;
        text-decoration: none;
        &:hover {
          color: $white;
          opacity: 100%;
        }
      }
    }
  }
}
</style>
