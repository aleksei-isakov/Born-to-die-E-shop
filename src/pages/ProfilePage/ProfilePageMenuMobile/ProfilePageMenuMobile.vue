<template>
  <div v-if="isMobile" class="profile-page__container">
    <div class="profile-page-menu__bottombar">
      <div v-for="item in items" :key="item.id" class="bottombar-menu-item">
        <base-button-router :path="item.path">
          <i :class="`${item.icon} bottombar-menu-item__icon fa-2x`" />
        </base-button-router>
        <base-button-router :path="item.path"
          >{{ item.title }}
        </base-button-router>
      </div>
    </div>
  </div>
</template>

<script>
import BaseButtonRouter from '@/base_components/BaseButtonRouter/BaseButtonRouter';
import { mapGetters } from 'vuex';

export default {
  name: 'ProfilePageMenuMobile',

  components: {
    BaseButtonRouter
  },

  data() {
    return {
      items: [
        {
          title: 'Profile',
          icon: 'fas fa-house-user',
          path: '/profile'
        },

        {
          title: 'Wishlist',
          icon: 'fas fa-heart',
          path: '/wishlist'
        },

        {
          title: 'Orders',
          icon: 'fas fa-folder',
          path: '/orders'
        },

        {
          title: 'Address book',
          icon: 'fas fa-address-book',
          path: '/address-book'
        }
      ]
    };
  },

  computed: {
    ...mapGetters('ProfilePageModule', ['isDesktop', 'isMobile'])
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.profile-page-menu {
  &__bottombar {
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    position: absolute;
    bottom: 0;
    width: 100%;
    background: $primary;

    > .bottombar-menu-item {
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
      width: 25%;
      opacity: 50%;
      min-height: 80px;

      .base-button-router {
        color: #ffffff;
        text-decoration: none;
        &:hover {
          color: #ffffff;
          opacity: 100%;
        }
      }

      &:hover {
        color: #ffffff;
        opacity: 100%;
      }
    }
  }
}
</style>
