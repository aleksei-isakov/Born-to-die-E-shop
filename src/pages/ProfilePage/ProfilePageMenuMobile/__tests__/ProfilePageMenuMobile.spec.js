import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import ProfilePageMenuMobile from '@/pages/ProfilePage/ProfilePageMenuMobile/ProfilePageMenuMobile';
import BaseButtonRouter from '@/base_components/BaseButtonRouter/BaseButtonRouter';

let state;
let wrapper;
let store;
let actions;
let getters;
const localVue = createLocalVue();

localVue.use(Vuex);

beforeEach(() => {
  getters = {
    isDesktop: () => false,
    isMobile: () => true
  };

  store = new Vuex.Store({
    modules: {
      ProfilePageModule: {
        actions,
        getters,
        state,
        namespaced: true
      }
    }
  });

  wrapper = shallowMount(ProfilePageMenuMobile, {
    stubs: {
      BaseButtonRouter: BaseButtonRouter
    },

    store,
    localVue
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe('ProfilePageMenuMobile', () => {
  it('renders the mobile version of ProfilePage', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain bottombar', () => {
    expect(wrapper.find('.profile-page-menu__bottombar').exists()).toBe(true);
  });
});
