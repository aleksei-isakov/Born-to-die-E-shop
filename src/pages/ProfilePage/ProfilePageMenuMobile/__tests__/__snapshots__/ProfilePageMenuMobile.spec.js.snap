// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePageMenuMobile renders the mobile version of ProfilePage 1`] = `
<div class="profile-page__container">
  <div class="profile-page-menu__bottombar">
    <div class="bottombar-menu-item">
      <router-link to="/profile" class="base-button-router"><i class="fas fa-house-user bottombar-menu-item__icon fa-2x"></i></router-link>
      <router-link to="/profile" class="base-button-router">Profile
      </router-link>
    </div>
    <div class="bottombar-menu-item">
      <router-link to="/wishlist" class="base-button-router"><i class="fas fa-heart bottombar-menu-item__icon fa-2x"></i></router-link>
      <router-link to="/wishlist" class="base-button-router">Wishlist
      </router-link>
    </div>
    <div class="bottombar-menu-item">
      <router-link to="/orders" class="base-button-router"><i class="fas fa-folder bottombar-menu-item__icon fa-2x"></i></router-link>
      <router-link to="/orders" class="base-button-router">Orders
      </router-link>
    </div>
    <div class="bottombar-menu-item">
      <router-link to="/address-book" class="base-button-router"><i class="fas fa-address-book bottombar-menu-item__icon fa-2x"></i></router-link>
      <router-link to="/address-book" class="base-button-router">Address book
      </router-link>
    </div>
  </div>
</div>
`;
